packages/application-shell/src/components/application-entry-point/application-entry-point.tsx

```diff
@@ -0,0 +1,58 @@
+import type { TProviderProps } from '@commercetools-frontend/application-shell-connectors';
+
+import React from 'react';
+import { Redirect, Route, Switch } from 'react-router-dom';
+import invariant from 'tiny-invariant';
+import RouteCatchAll from '../route-catch-all';
+
+type Props<AdditionalEnvironmentProperties extends {}> = {
+  environment: TProviderProps<AdditionalEnvironmentProperties>['environment'];
+  render?: () => JSX.Element;
+  children?: React.ReactNode;
+};
+
+const ApplicationEntryPoint = <AdditionalEnvironmentProperties extends {}>(
+  props: Props<AdditionalEnvironmentProperties>
+) => {
+  // If the `children` prop is used (instead of the `render` prop),
+  // we pre-configure the application entry point routes to avoid
+  // users to do so on their own.
+  if (props.children) {
+    const entryPointUriPath = props.environment.entryPointUriPath;
+    return (
+      <Switch>
+        {
+          // For development, it's useful to redirect to the actual
+          // application routes when you open the browser at http://localhost:3001
+          process.env.NODE_ENV === 'production' ? null : (
+            <Redirect
+              exact={true}
+              from="/:projectKey"
+              to={`/:projectKey/${entryPointUriPath}`}
+            />
+          )
+        }
+        <Route path={`/:projectKey/${entryPointUriPath}`}>
+          {props.children}
```

Comment:
We could use React.Children.only<React.ReactNode>(props.children);` to verify that children has only one child (a React element) and return it otherwise throw an error.